### PR TITLE
feat: New Module `avm/res/relay/namespace`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/relay/namespace`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.relay.namespace](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.relay.namespace.yml) |

### CODE REVIEW

File: avm/res/relay/namespace/wcf-relay/authorization-rule/version.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
+    "version": "0.4",
```
Comment (AlexanderSehr) [inline avm/res/relay/namespace/wcf-relay/authorization-rule/version.json:3]:
```suggestion
    "version": "0.1",
```

File: avm/res/relay/namespace/main.bicep [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+}
+
+resource namespace_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for (diagnosticSetting, index) in (diagnosticSettings ?? []): {
+  name: diagnosticSetting.?name ?? '${name}-diagnosticSettings'
+  properties: {
+    storageAccountId: diagnosticSetting.?storageAccountResourceId
+    workspaceId: diagnosticSetting.?workspaceResourceId
+    eventHubAuthorizationRuleId: diagnosticSetting.?eventHubAuthorizationRuleResourceId
+    eventHubName: diagnosticSetting.?eventHubName
+    metrics: diagnosticSetting.?metricCategories ?? [
```
Comment (AlexanderSehr) [inline avm/res/relay/namespace/main.bicep:200]: Please update this resource to the latest spec

File: avm/res/relay/namespace/main.bicep [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+  @description('Optional. Tags to be applied on all resources/resource groups in this deployment.')
+  tags: object?
+
+  @description('Optional. Manual PrivateLink Service Connections.')
+  manualPrivateLinkServiceConnections: array?
+
+  @description('Optional. Enable/Disable usage telemetry for module.')
+  enableTelemetry: bool?
+}[]?
```
Comment (AlexanderSehr) [inline avm/res/relay/namespace/main.bicep:391]:
```suggestion

  @description('Optional. Specify if you want to deploy the Privte Endpoint into a different resource group than the main resource.')
  resourceGroupName: string?
}[]?
```

File: avm/res/relay/namespace/main.bicep [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+  condition: string?
+
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
+}[]?
+
+type privateEndpointType = {
```
Comment (AlexanderSehr) [inline avm/res/relay/namespace/main.bicep:325]: Please update to latest schema. Is missing the new manual connection boolean etc.

File: avm/res/relay/namespace/tests/e2e/waf-aligned/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    skuName: 'Standard'
```
Comment (AlexanderSehr) [inline avm/res/relay/namespace/tests/e2e/waf-aligned/main.test.bicep:68]: Required for WAF-alignment?

File: avm/res/relay/namespace/main.bicep [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+    privateDnsZoneGroupName: privateEndpoint.?privateDnsZoneGroupName
+    privateDnsZoneResourceIds: privateEndpoint.?privateDnsZoneResourceIds
+    roleAssignments: privateEndpoint.?roleAssignments
+    tags: privateEndpoint.?tags ?? tags
+    customDnsConfigs: privateEndpoint.?customDnsConfigs
+    ipConfigurations: privateEndpoint.?ipConfigurations
+    applicationSecurityGroupResourceIds: privateEndpoint.?applicationSecurityGroupResourceIds
+    customNetworkInterfaceName: privateEndpoint.?customNetworkInterfaceName
+  }
+}
```
Comment (AlexanderSehr) [inline avm/res/relay/namespace/main.bicep:260]:
```suggestion
module namespace_privateEndpoints 'br/public:avm/res/network/private-endpoint:0.4.1' = [for (privateEndpoint, index) in (privateEndpoints ?? []): {
  name: '${uniqueString(deployment().name, location)}-namespace-PrivateEndpoint-${index}'
  params: {
    name: privateEndpoint.?name ?? 'pep-${last(split(namespace.id, '/'))}-${privateEndpoint.?service ?? 'namespace'}-${index}'
    privateLinkServiceConnections: privateEndpoint.?isManualConnection != true ? [
      {
        name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(namespace.id, '/'))}-${privateEndpoint.?service ?? 'namespace'}-${index}'
        properties: {
          privateLinkServiceId: namespace.id
          groupIds: [
            privateEndpoint.?service ?? 'namespace'
          ]
        }
      }
    ] : null
    isManualConnection: privateEndpoint.?isManualConnection == true ? [
      {
        name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(namespace.id, '/'))}-${privateEndpoint.?service ?? 'namespace'}-${index}'
        properties: {
          privateLinkServiceId: namespace.id
          groupIds: [
            privateEndpoint.?service ?? 'namespace'
          ]
          requestMessage: privateEndpoint.?manualConnectionRequestMessage ?? 'Manual approval required.'
        }
      }
    ] : null
    subnetResourceId: privateEndpoint.subnetResourceId
    enableTelemetry: privateEndpoint.?enableTelemetry ?? enableTelemetry
    location: privateEndpoint.?location ?? reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
    lock: privateEndpoint.?lock ?? lock
    privateDnsZoneGroupName: privateEndpoint.?privateDnsZoneGroupName
    privateDnsZoneResourceIds: privateEndpoint.?privateDnsZoneResourceIds
    roleAssignments: privateEndpoint.?roleAssignments
    tags: privateEndpoint.?tags ?? tags
    customDnsConfigs: privateEndpoint.?customDnsConfigs
    ipConfigurations: privateEndpoint.?ipConfigurations
    applicationSecurityGroupResourceIds: privateEndpoint.?applicationSecurityGroupResourceIds
    customNetworkInterfaceName: privateEndpoint.?customNetworkInterfaceName
  }
  scope: resourceGroup(privateEndpoint.?resourceGroupName ?? '')
}
```

File: avm/res/relay/namespace/tests/e2e/waf-aligned/main.test.bicep (hunk 5 above)
Comment (krbar) [reply avm/res/relay/namespace/tests/e2e/waf-aligned/main.test.bicep]: I removed it as it is the default value: https://github.com/Azure/bicep-registry-modules/pull/1319/commits/14f67236ec2d1bb9aa8d7fe88274cfe9d0f784a6

File: avm/res/relay/namespace/main.bicep (hunk 4 above)
Comment (krbar) [reply avm/res/relay/namespace/main.bicep:317]: Done: https://github.com/Azure/bicep-registry-modules/pull/1319/commits/9f9c82d23b6aa9322f1837b44c87af20d8b1d4c6

File: avm/res/relay/namespace/main.bicep (hunk 2 above)
Comment (krbar) [reply avm/res/relay/namespace/main.bicep]: Done: https://github.com/Azure/bicep-registry-modules/pull/1319/commits/9f9c82d23b6aa9322f1837b44c87af20d8b1d4c6

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**